go/vt/vtgate/planbuilder/operators/SQL_builder.go

```diff
@@ -566,21 +566,16 @@ func buildProjection(op *Projection, qb *queryBuilder) error {
 func buildApplyJoin(op *ApplyJoin, qb *queryBuilder) error {
 	predicates := slice.Map(op.JoinPredicates, func(jc JoinColumn) sqlparser.Expr {
 		// since we are adding these join predicates, we need to mark to broken up version (RHSExpr) of it as done
-		qb.ctx.SkipPredicates[jc.RHSExpr] = nil
-
+		_ = qb.ctx.SkipJoinPredicates(jc.Original.Expr)
 		return jc.Original.Expr
 	})
+
```

Comment:
the error returned is not checked.
Should this be handled?

Reply:
I think we talked about this with @systay, and handling the error leads to some changes in behaviour and various tests failures. 🤔

Reply:
It was for Foreign keys but that should not happen anymore.

Reply:
I've added the error checking. let's see what CI thinks 🤞